src/Queue/ClientJob.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace CloudCreativity\LaravelJsonApi\Queue;
+
+use Carbon\Carbon;
+use CloudCreativity\LaravelJsonApi\Api\Api;
+use CloudCreativity\LaravelJsonApi\Contracts\Queue\AsynchronousProcess;
+use CloudCreativity\LaravelJsonApi\Exceptions\RuntimeException;
+use CloudCreativity\LaravelJsonApi\Object\ResourceIdentifier;
+use Illuminate\Database\Eloquent\Model;
+use Ramsey\Uuid\Uuid;
+
+class ClientJob extends Model implements AsynchronousProcess
+{
+
+    /**
+     * @var bool
+     */
+    public $incrementing = false;
+
+    /**
+     * @var string
+     */
+    protected $table = 'json_api_client_jobs';
+
+    /**
+     * @var string
+     */
+    protected $primaryKey = 'uuid';
+
+    /**
+     * Mass-assignable attributes.
+     *
+     * @var array
+     */
+    protected $fillable = [
+        'api',
+        'attempts',
+        'completed_at',
+        'failed',
+        'resource_type',
+        'resource_id',
+        'timeout',
+        'timeout_at',
+        'tries',
+    ];
+
+    /**
+     * Default attributes.
+     *
+     * @var array
+     */
+    protected $attributes = [
+        'failed' => false,
+        'attempts' => 0,
+    ];
+
+    /**
+     * @var array
+     */
+    protected $casts = [
+        'attempts' => 'integer',
+        'failed' => 'boolean',
+        'timeout' => 'integer',
+        'tries' => 'integer',
+    ];
+
+    /**
+     * @var array
+     */
+    protected $dates = [
+        'completed_at',
+        'timeout_at',
+    ];
+
+    /**
+     * @inheritdoc
+     */
+    public static function boot()
+    {
+        parent::boot();
+
+        static::addGlobalScope(new ClientJobScope());
+
+        static::creating(function (ClientJob $job) {
+            $job->uuid = $job->uuid ?: Uuid::uuid4()->toString();
```

Comment:
UUID dependency needs to be added to the `composer.json` as we now have a hard dependency on it.